patches/tModLoader/Terraria/GameContent/UI/EmoteBubble.cs.patch

```diff
@@ -1,11 +1,35 @@
 --- src/Terraria/Terraria/GameContent/UI/EmoteBubble.cs
 +++ src/tModLoader/Terraria/GameContent/UI/EmoteBubble.cs
-@@ -9,7 +_,7 @@
+@@ -4,12 +_,13 @@
+ using System.Collections.Generic;
+ using Terraria.GameContent.Events;
+ using Terraria.ID;
++using Terraria.ModLoader;
+ 
+ namespace Terraria.GameContent.UI
  {
  	public class EmoteBubble
  	{
 -		private static int[] CountNPCs = new int[663];
-+		internal static int[] CountNPCs = new int[NPCID.Count];
++		internal static int[] CountNPCs = new int[NPCLoader.NPCCount];
```

Comment:
This won't work, because `NPCCount` will change when mods are loaded and unloaded. You need to reference this in `NPCLoader.ResizeArrays`. Keep this in mind for any arrays active when mods are loaded and do the same for ItemLoader and etc. The only time this will work is if it's a non-static field and the array is constructed and destroyed during gameplay. (that is, when the array lifetime doesn't overlap with mod loading)